tokio/src/macros/select.rs

```diff
@@ -489,13 +489,22 @@ doc! {macro_rules! select {
         // Create a scope to separate polling from handling the output. This
         // adds borrow checker flexibility when using the macro.
         let mut output = {
+            // Store each future directly first (that is, without wrapping the future in a call to
+            // `IntoFuture::into_future`). This allows the `$fut` expression to make use of
+            // temporary lifetime extension.
+            //
+            // https://doc.rust-lang.org/1.58.1/reference/destructors.html#temporary-lifetime-extension
+            let futures_init = ($( $fut, )+);
+
             // Safety: Nothing must be moved out of `futures`. This is to
             // satisfy the requirement of `Pin::new_unchecked` called below.
             //
             // We can't use the `pin!` macro for this because `futures` is a
             // tuple and the standard library provides no way to pin-project to
             // the fields of a tuple.
-            let mut futures = ( $( $crate::macros::support::IntoFuture::into_future($fut) , )+ );
+            let mut futures = ($( $crate::macros::support::IntoFuture::into_future(
+                        $crate::count_field!( futures_init.$($skip)* )
```

Comment:
Should we delete some indentations here?

Reply:
Shrug.